### PR TITLE
LibRegex: Make infinite repetitions short-circuit on empty matches :infinity::infinity::infinity:

### PR DESCRIPTION
**Pull request by alimpfard**
This makes (addmittedly weird) patterns like `(a*)*` work correctly
without going into an infinite fork loop.

:infinity::infinity::infinity::infinity::infinity::infinity::infinity:

Starring: I hate offset math!

### CODE REVIEW

File: Userland/Libraries/LibRegex/RegexByteCode.cpp [hunk 1]
```diff
@@ -144,60 +160,14 @@ void ByteCode::ensure_opcodes_initialized()
...
-            s_opcodes[i] = make<OpCode_ResetRepeat>();
-            break;
+#define __ENUMERATE_OPCODE(OpCode)              \
+    case OpCodeId::OpCode:                      \
+        s_opcodes[i] = make<OpCode_##OpCode>(); \
+        break;
+
+            ENUMERATE_OPCODES
+
+#undef __ENUMERATE_OPCODE
```
Comment (linusg) [inline Userland/Libraries/LibRegex/RegexByteCode.cpp:170]: sneaky refactoring in an unrelated commit

**Review by linusg — APPROVED** [review]
LGTM, thanks for fixing!

File: Userland/Libraries/LibRegex/RegexByteCode.cpp (hunk 1 above)
Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexByteCode.cpp:170]: I got _really_ upset that I had to add 3 more lines, so I added 8 more lines instead.
